station-worldgen-api-v0/src/main/java/net/modificationstation/stationapi/api/worldgen/biome/BiomeBuilder.java

```diff
@@ -0,0 +1,195 @@
+package net.modificationstation.stationapi.api.worldgen.biome;
+
+import net.minecraft.level.biome.Biome;
+import net.modificationstation.stationapi.api.worldgen.surface.SurfaceRule;
+import net.modificationstation.stationapi.impl.worldgen.BiomeColorsImpl;
+
+import javax.swing.text.html.parser.Entity;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class BiomeBuilder {
+	private static final BiomeBuilder INSTANCE = new BiomeBuilder();
+	private final Map<Class<? extends Entity>, Integer> hostileEntities = new HashMap<>();
+	private final Map<Class<? extends Entity>, Integer> passiveEntities = new HashMap<>();
+	private final Map<Class<? extends Entity>, Integer> waterEntities = new HashMap<>();
```

Comment:
Reference2IntMap would be better here. Classes can be safely compared by identity and int won't be autoboxed.

Reply:
In vanilla biomes classes are used directly, so I used them here too, this is just a temporal storage. And such format allows to use some nice sugar like this: `hostileEntities.forEach(biome::addHostileEntity);`

Reply:
Replaces, seems to work fine